confgenerator/confgenerator.go

```diff
@@ -57,20 +57,24 @@ func googleCloudExporter(userAgent string, instrumentationLabels bool, serviceRe
 	}
 }
 
-func ConvertToOtlpExporter(receiver otel.ReceiverPipeline, ctx context.Context) otel.ReceiverPipeline {
+func ConvertToOtlpExporter(pipeline otel.ReceiverPipeline, ctx context.Context, systemMetric bool) otel.ReceiverPipeline {
 	expOtlpExporter := experimentsFromContext(ctx)["otlp_exporter"]
 	resource, _ := platform.FromContext(ctx).GetResource()
 	if !expOtlpExporter {
-		return receiver
+		return pipeline
 	}
-	_, err := receiver.ExporterTypes["metrics"]
+	_, err := pipeline.ExporterTypes["metrics"]
 	if !err {
-		return receiver
+		return pipeline
 	}
-	receiver.ExporterTypes["metrics"] = otel.OTLP
+	pipeline.ExporterTypes["metrics"] = otel.OTLP
 
-	receiver.Processors["metrics"] = append(receiver.Processors["metrics"], otel.GCPProjectID(resource.ProjectName()))
-	return receiver
+	pipeline.Processors["metrics"] = append(pipeline.Processors["metrics"], otel.GCPProjectID(resource.ProjectName()))
+	if systemMetric {
+		pipeline.Processors["metrics"] = append(pipeline.Processors["metrics"], otel.MetricStartTime())
```

Comment:
Is there a "generic" way to set `MetricStartTime` and `MetricsRemoveInstrumentationLibraryLabelsAttributes` that works for all metrics without needing to differentiate explicitly using an extra parameter like `systemMetric` ?

Reply:
No, this is the equivalent we have today for the different ways we use the GCM exporter. We call is system when we remove the instrumentation labels attributes, and OTEL when we don't. This is just mimicking that same behavior. It's really an XOR behavior here. For the metricstarttime we can just add it everywhere when using the otlphttp exporter